en/net/advanced-operations/securing-and-signing/sign-pdf-document-from-smart-card/_index.md

```diff
@@ -137,40 +154,54 @@ using (Document document = new Document(dataDir + "externalSignature1.pdf"))
 ## Sign With Smart Card Using PDF File Signature
 
 ```csharp
-// For complete examples and data files, please go to https://github.com/aspose-pdf/Aspose.PDF-for-.NET
-// The path to the documents directory.
-string dataDir = RunExamples.GetDataDir_AsposePdf_SecuritySignatures();
+// For complete examples and data files, check for https://github.com/aspose-pdf/Aspose.PDF-for-.NET
 
-using (Document document = new Document(dataDir + "blank.pdf"))
+private void SignWithSmartCard()
 {
-    using (Facades.PdfFileSignature pdfSign = new Facades.PdfFileSignature())
+    // The path to the documents directory
+    string dataDir = RunExamples.GetDataDir_AsposePdf_SecuritySignatures();
+    
+    // Open the document
+    using (var document = new Aspose.Pdf.Document(dataDir + "blank.pdf"))
     {
-        pdfSign.BindPdf(document);
+        using (var pdfSign = new Aspose.Pdf.Facades.PdfFileSignature())
+        {   
+            pdfSign.BindPdf(document);
 
-        // Sign with certificate selection in the windows certificate store
-        System.Security.Cryptography.X509Certificates.X509Store store = new System.Security.Cryptography.X509Certificates.X509Store(System.Security.Cryptography.X509Certificates.StoreLocation.CurrentUser);
-        store.Open(System.Security.Cryptography.X509Certificates.OpenFlags.ReadOnly);
-        // Manually chose the certificate in the store
-        System.Security.Cryptography.X509Certificates.X509Certificate2Collection sel = System.Security.Cryptography.X509Certificates.X509Certificate2UI.SelectFromCollection(store.Certificates, null, null, System.Security.Cryptography.X509Certificates.X509SelectionFlag.SingleSelection);
-
-        ExternalSignature externalSignature = new ExternalSignature(sel[0]);
-        pdfSign.SignatureAppearance = dataDir + "demo.png";
-        pdfSign.Sign(1, "Reason", "Contact", "Location", true, new System.Drawing.Rectangle(100, 100, 200, 200), externalSignature);
-        // Save result file
-        pdfSign.Save(dataDir + "externalSignature2.pdf");
+            // Sign with certificate selection in the windows certificate store
+            X509Store store = new X509Store(StoreLocation.CurrentUser);
+            store.Open(OpenFlags.ReadOnly);
+            // Manually chose the certificate in the store
+            X509Certificate2Collection sel = X509Certificate2UI.SelectFromCollection(store.Certificates, null, null, X509SelectionFlag.SingleSelection);
```

Comment:
We may not specify full qualified name for the System libraries. However, `System.Security.Cryptography.X509Certificates` is enough hard case to specify it.
Additionally `var` can be used.